src/pybind/mgr/dashboard/frontend/src/app/shared/api/pool.service.ts

```diff
@@ -35,8 +38,8 @@ export class PoolService {
     return this.http.get(this.apiPath);
   }
 
-  getInfo() {
-    return this.http.get(`${this.apiPath}/_info`);
+  getInfo(): Observable<PoolFormInfo> {
```

Comment:
I'm not sure about adding the return type for every API request, because we haven't done that as far a I know, at least not that explicitly. I'm not in favor of having that.

Reply:
Let me try to address your concerns. As far as I know now, it doesn't make sense to enforce it everywhere. But it does make much sense to do it. Every time you use a service, you'll benefit from knowing the type of the result. The IDE will know it, too. It will then infer the type from every subsequent use of the result and provide you with code completion, inspections and refactoring capabilities along the way. I'd tend to say it's a very important position in the code to have the return type in.

Reply:
Could you than make a separate PR to address all services to do so?

Reply:
Why does it make sense here than? As this service method is only used in the pool form component, furthermore it should be only used there.